### DISCUSSION TITLE
x-consumer-timeout can not be set individually per consumer

### DISCUSSION THREAD

**Discussion by labankallgren** [rabbitmq/rabbitmq-server#9268] [Questions]
### Describe the bug

There is a declared queue with one publisher and several consumers.
When the consumer declares the queue it passes the optional queue argument `x-consumer-timeout` as described in https://www.rabbitmq.com/consumers.html.
The consumer timeout only kicks in if no queue with that name exists, i. e. the queue is declared for the first time.
If the queue already has been declared (exists) the consumer timeout is not reset.

There are two bugs here:
Primary: The consumers of a specific queue cannot individually set a consumer timeout
Secondary: There is no error response that the attempt to set the consumer timeout on a queue failed (compare with setting x-max-priority), 

### Reproduction steps

1. Let one consumer declare a queue with no x-consumer-timeout optional argument set
2. Let another consume declare the same queue with x-consumer-timeout set
...

### Expected behavior

1. The consumer timeout is default for the first consumer
2. The consumer timeout is the desired x-consumer-timeout for the second consumer

### Additional context

This is tested on rabbitmq server 3.13.0-beta

**Comment by lukebakken** [comment]
> The consumer timeout only kicks in if no queue with that name exists, i. e. the queue is declared for the first time.

Yes, this is how _all_ queue arguments work. Once set, you can't change them.

> Primary: The consumers of a specific queue cannot individually set a consumer timeout

This probably won't be implemented, but I'll let @michaelklishin chime in.

> Secondary: There is no error response that the attempt to set the consumer timeout on a queue failed (compare with setting x-max-priority)

That's surprising, and seems like a bug ... @michaelklishin?

**Reply by michaelklishin** [reply]
How do we define "failure"? Please put together an executable example.

**Reply by lukebakken** [reply]
This would be a `PRECONDITION FAILED` error like we see with other `x-` queue aruguments.
